core/examples/wasm.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::op;
+use deno_core::Extension;
+use deno_core::JsRuntime;
+use deno_core::RuntimeOptions;
+use std::ptr::NonNull;
+use std::mem::transmute;
+
+// This is a hack to make the `#[op]` macro work with
+// deno_core examples.
+// You can remove this:
+
+use deno_core::*;
+
+struct WasmMemory(NonNull<v8::WasmMemoryObject>);
+
+unsafe fn wasm_memory_unchecked(state: &mut OpState) -> &mut [u8] {
+  let WasmMemory(global) = state.borrow::<WasmMemory>();
+  // SAFETY: `v8::Local` is always non-null pointer; the `HandleScope` is
+  // already on the stack, but we don't have access to it.
+  let memory_object = unsafe {
+    transmute::<NonNull<v8::WasmMemoryObject>, v8::Local<v8::WasmMemoryObject>>(
+      *global,
+    )
+  };
+  let backing_store = memory_object.buffer().get_backing_store();
+  let ptr = backing_store.data().unwrap().as_ptr() as *mut u8;
+  let len = backing_store.byte_length();
+  // SAFETY: `ptr` is a valid pointer to `len` bytes.
+  unsafe { std::slice::from_raw_parts_mut(ptr, len) }
```

Comment:
Is it possible that ptr is a null pointer? eg. Wasm module created with 0 length memory buffer?

Reply:
@littledivy what's the answer here?

Reply:
Quick test shows that it is not nullptr for an empty memory buffer.

Reply:
Maybe it can be...idk...this is just an example